jupyter_archive/handlers.py

```diff
@@ -229,6 +229,18 @@ def extract_archive(self, archive_path):
         self.log.info("Begin extraction of {} to {}.".format(archive_path, archive_destination))
 
         archive_reader = make_reader(archive_path)
+
+        if isinstance(archive_reader, tarfile.TarFile):
+            # Check file path to avoid path traversal
+            # See https://nvd.nist.gov/vuln/detail/CVE-2007-4559
+            with archive_reader as archive:
+                for name in archive_reader.getnames():
+                    if os.path.relpath(archive_destination / name, archive_destination).startswith("../"):
```

Comment:
I would use `os.pardir` directly for readability and because the name can be `'../test'` => relpath == `".."`

```suggestion
                    if os.path.relpath(archive_destination / name, archive_destination).startswith(os.pardir):
```

Reply:
I agree with your suggestion, thank you.